jdaviz/app.py

```diff
@@ -95,13 +107,25 @@ def to_unit(self, data, cid, values, original_units, target_units):
         # gives the units of the values array, which might not be the same
         # as the original native units of the component in the data.
         if cid.label == "flux":
+            # to debug
+            # print('orginal_units : ', original_units, "target_units : ", target_units)
+
             spec = data.get_object(cls=Spectrum1D)
             if len(values) == 2:
                 # Need this for setting the y-limits
                 spec_limits = [spec.spectral_axis[0].value, spec.spectral_axis[-1].value]
                 eqv = u.spectral_density(spec_limits*spec.spectral_axis.unit)
+
+            elif '_pixel_scale_factor' in spec.meta:
```

Comment:
Check if this condition ever gets triggered for you - it does not appear to be to me, which I think is the root cause of the issues you are seeing.

Reply:
The pixel_scale_factor is set to spectrum that pass through Cubeviz's collapse function. Another effort that is being worked on is to have all spectra go through the collapse function instead of glues auto-collapse. When this is implemented all spectra that appear in the spectrum-viewer will have this pixel scale factor. 

For the purpose of this PR, we do need to create a spatial subset, and pass it through spectral extraction in order to have a pixel scale factor stored. Once this is completed, and the translation is attempted, it will enter this elif statement. It does also mean that for now, only this one spectrum will have translate in the viewer.

Reply:
In tagup today, Kyle suggested something similar. I plan on testing this PR where I only use the spectra that have passed through the collapse function (either removing the others or rebasing on Kyle's PR). I will get back to you soon on the results!